caas/kubernetes/provider/application/application.go

```diff
@@ -1789,9 +1789,13 @@ func (a *app) ApplicationPodSpec(config caas.ApplicationConfig) (*corev1.PodSpec
 			},
 		},
 	}
-	err := ApplyConstraints(spec, a.name, config.Constraints, configureConstraint)
-	if err != nil {
-		return nil, errors.Annotate(err, "processing constraints")
+
+	if err := ApplyWorkloadConstraints(spec, a.name, config.Constraints, configureWorkloadConstraint); err != nil {
+		return nil, errors.Annotate(err, "processing workload container constraints")
+	}
+
+	if err := ApplyCharmConstraints(spec, a.name, config.CharmConstraints, configureCharmConstraint); err != nil {
+		return nil, errors.Annotate(err, "processing charm container constraints")
 	}
```

Comment:
I'm not sure this abstraction is needed here since we are only setting memory request/limit. We can keep it simple and refactor later if ever needed. So we don't need `configureCharmConstraint` and can just implement the logic directly in `ApplyCharmConstraints`